tests/integ/sagemaker/workflow/test_workflow.py

```diff
@@ -1019,6 +1020,99 @@ def test_create_and_update_with_parallelism_config(
             pass
 
 
+def test_create_and_start_without_parallelism_config_override(
+    pipeline_session, role, pipeline_name, script_dir
+):
+    sklearn_train = SKLearn(
+        framework_version="0.20.0",
+        entry_point=os.path.join(script_dir, "train.py"),
+        instance_type="ml.m5.xlarge",
+        sagemaker_session=pipeline_session,
+        role=role,
+    )
+
+    train_steps = [
+        TrainingStep(
+            name=f"my-train-{count}",
+            display_name="TrainingStep",
+            description="description for Training step",
+            step_args=sklearn_train.fit(),
+        )
+        for count in range(2)
+    ]
+    pipeline = Pipeline(
+        name=pipeline_name,
+        steps=train_steps,
+        sagemaker_session=pipeline_session,
+    )
+
+    try:
+        pipeline.create(role, parallelism_config=dict(MaxParallelExecutionSteps=1))
+        # No ParallelismConfiguration given in pipeline.start, so it won't override that in pipeline.create
+        execution = pipeline.start()
+
+        def validate():
+            # Only one step would be scheduled initially
+            assert len(execution.list_steps()) == 1
```

Comment:
Just clarifying here -- do we need to check for the `Executing` status here as well? I may be missing what exactly list_steps() is doing. Just taking a brief look at our backend logic, Im seeing that we simply return a page of max 100 results [[link](https://code.amazon.com/packages/IronmanTiogaServiceObjects/blobs/b1b68ec97fe2a8dd94a8be0dd7ae7d459d25ec51/--/src/com/amazon/ironmantiogaservice/datastore/requests/ListPipelineExecutionStepsRequest.java#L20)]

Why would this not be 2, with one in executing status and the other in, I'd assume, `PENDING` or `AVAILABLE_TO_RUN` status?

Reply:
Ok I see. So our list response will iteratively grow in size as the execution runs, such that dependent steps will get created as StepEntities in our DDB, and thus show up in the `list_steps()` response page.  Thanks!